### ISSUE TITLE
BUG: Urgent: Script Functionality Breakdown in Excalidraw 2.0.19 with Text in Rectangles

### ISSUE THREAD

**Issue by wwjj12345** [zsviczian/obsidian-excalidraw-plugin#1576]
I've encountered several bugs in the scripts of the Excalidraw plugin for Obsidian after updating to the latest version 2.0.19. Here are the details:

When using the Rectangle tool (shortcut key 2) for drawing, for instance, after adding two Rectangle elements, functions like Auto Layout, Connect elements, and Expand rectangles horizontally work fine as long as no text is added to the rectangles. However, as soon as any text (letters, numbers, etc.) is added to a Rectangle element, all these functions (Auto Layout, Connect elements, Expand rectangles horizontally) cease to work. Deleting all the text from the rectangles restores the functionality of these features.
Additionally, I was in the process of using Excalidraw for note-taking when I upgraded to version 2.0.19. I can assure you that before the upgrade, adding text to Rectangle elements never interfered with any script functionalities, as I had been continuously utilizing these script features prior to the upgrade.

My computer is a MAC sonoma 14.2.1. I've tried reinstalling Excalidraw and its scripts, but the problem persists.

**Comment by zsviczian** [comment]
Under the hood the Excalidraw package made some changes to how containers are linked to elements. I tested the update on myself for a week before releasing it. I don't personally use the script you mention, to the point, that those were contributed by the community, I did not develop them. It seems these scripts make use of the container functionality. I will test to see what's going on.

Until then possibly the best strategy for you is to downgrade to 2.0.18. I don't promise I will work on this this weekend.

**Comment by wwjj12345** [comment]
Thank you very much for your prompt response and for looking into the issue. I understand the complexities involved with integrating community-contributed scripts and appreciate your efforts in maintaining the Excalidraw package. Downgrading to version 2.0.18 seems like a reasonable temporary solution, and I'll proceed with that.

**Comment by zsviczian** [comment]
Can I ask you to test with this version?
Download and unzip. Replace main.js with this in the .obsidian/plugins/obsidian-excalidraw-plugin folder
[main.js.zip](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/14155778/main.js.zip)

**Comment by wwjj12345** [comment]
I followed your instructions and replaced the main.js file in the .obsidian/plugins/obsidian-excalidraw-plugin folder with the version you provided. I'm thrilled to report that the bug has been fixed! The Auto Layout, Connect elements, and Expand rectangles horizontally scripts are all working perfectly now!

I want to express my sincere gratitude for your prompt and effective solution. Your support and dedication to resolving this issue have greatly enhanced my experience with Excalidraw in Obsidian.
